src/adapter/controller/task.go

```diff
@@ -0,0 +1,86 @@
+package controller
+
+import (
+	"database/sql"
+	"encoding/json"
+	"github.com/liampm/static-code-analysis-tool/domain"
+	uuid "github.com/satori/go.uuid"
+	"net/http"
+)
+
+type TaskController struct {
+	ReadRepo  domain.TaskReadRepo
+	WriteRepo domain.TaskRepo
+}
+
+func (controller *TaskController) All() func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+
+		projectUuid, err := uuidFromParam(r, "projectId")
+		if err != nil {
+			w.WriteHeader(404) // Not found for any invalid IDs
+			return
+		}
+
+		err = marshalJSONResponse(w, controller.ReadRepo.AllForProject(projectUuid))
+		if err != nil {
+			panic(err) // Panic whilst in development
+		}
+
+		w.WriteHeader(200)
```

Comment:
`2019/05/24 10:27:56 http: superfluous response.WriteHeader call from github.com/liampm/static-code-analysis-tool/adapter/controller.(*TaskController).All.func1 (task.go:31)`

Not sure if we need this then? Is it handled by the caller?

Reply:
It should be moved above the calls to write (done by marshalJSONResponse)
It's harmless in this instance but should move